### PR TITLE
[MMCA-4457] - Remove unused code and ToDos through out the services 

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - Code changes for Live services handover

Below has been covered in this PR

- remove scalastyle warnings from code and tests
- remove compile time warnings
- remove unused code
- use of constants instead of hard coded values
- new test classes
- add new test cases
- update sbt-auto-build to 3.20.0

### CODE REVIEW

File: app/controllers/SDESNotificationsController.scala [hunk 1]
```diff
@@ -32,7 +32,9 @@ import scala.concurrent.ExecutionContext
 class SDESNotificationsController @Inject()(notificationCache: NotificationCache,
                                             authorisedRequest: AuthorisedRequest,
                                             dateTimeService: DateTimeService,
-                                            cc: ControllerComponents)(implicit ec: ExecutionContext) extends BackendController(cc) with ControllerChecks {
+                                            cc: ControllerComponents)
+                                           (implicit ec: ExecutionContext)
```
Comment (AdamCoops) [inline app/controllers/SDESNotificationsController.scala:36]: I think the implicit can be moved to the end of the params here?

File: app/controllers/SearchAuthoritiesController.scala [hunk 2]
```diff
@@ -25,16 +25,16 @@ import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 import javax.inject.Inject
 import scala.concurrent.ExecutionContext
 
-class SearchAuthoritiesController @Inject()(
-                                             acc40Connector: Acc40Connector,
-                                             cc: ControllerComponents
+class SearchAuthoritiesController @Inject()(acc40Connector: Acc40Connector,
+                                            cc: ControllerComponents
                                            )(implicit ec: ExecutionContext) extends BackendController(cc) {
```
Comment (AdamCoops) [inline app/controllers/SearchAuthoritiesController.scala:30]: if we can't move the implicit up we can make the bracket placement consistent so the ) is on the controllerComponents line?

File: app/controllers/StatementSearchFailureNotificationController.scala [hunk 3]
```diff
@@ -39,14 +39,13 @@ import javax.inject.Inject
...
-                                                              histDocumentService: HistoricDocumentService,
-                                                              smc: SecureMessageConnector
+class StatementSearchFailureNotificationController @Inject()(cc: ControllerComponents,
+                                                             jsonSchemaValidator: JSONSchemaValidator,
+                                                             authorizationHeaderFilter: AuthorizationHeaderFilter,
+                                                             mdgHeaderFilter: MdgHeaderFilter,
+                                                             cacheService: HistoricDocumentRequestSearchCacheService,
+                                                             histDocumentService: HistoricDocumentService,
+                                                             smc: SecureMessageConnector
                                                             )(implicit execution: ExecutionContext)
```
Comment (AdamCoops) [inline app/controllers/StatementSearchFailureNotificationController.scala:49]: same here

File: app/controllers/StatementSearchFailureNotificationController.scala [hunk 4]
```diff
@@ -155,20 +139,16 @@ class StatementSearchFailureNotificationController @Inject()(
...
-  /**
-   * Checks the failureReasonCode and proceeds as below
-   *   If failureReasonCode is NoDocumentsFound, update the retry count
-   *   else process the RequestId and update the HistoricDocumentRequestSearch document
-   */
   private def checkFailureReasonAndProcessRequestId(statementRequestID: String,
                                                     correlationId: String,
                                                     failureReasonCode: String,
                                                     optHistDocReq: Option[HistoricDocumentRequestSearch]
                                                    )(implicit hc: HeaderCarrier): Future[Result] = {
```
Comment (AdamCoops) [inline app/controllers/StatementSearchFailureNotificationController.scala:146]: and here

File: app/models/responses/StatementSearchFailureNotificationErrorResponse.scala [hunk 5]
```diff
@@ -58,17 +58,19 @@ object StatementSearchFailureNotificationErrorResponse {
                                    errorDetailMsg: String,
                                    errorMsgList: Seq[String]): Seq[String] = {
     statementRequestID.fold(errorMsgList)(stReqId => Seq(
-      if (errorCode == ErrorCode.code500)
-        if (errorDetailMsg.isEmpty) ErrorMessage.technicalErrorDetail(stReqId)
-        else errorDetailMsg
-      else ErrorMessage.invalidStatementReqIdDetail(stReqId))
+      if (errorCode == ErrorCode.code500) {
```
Comment (AdamCoops) [inline app/models/responses/StatementSearchFailureNotificationErrorResponse.scala:61]: I think we can simplify this by using an if, else if, else statement.
Like if (code500 & is empty) do technicalerrordetails else if isEmpty do errordetailsmsg, else invalidstatemnet

File: app/services/NotificationCache.scala [hunk 6]
```diff
@@ -35,9 +35,9 @@ import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class DefaultNotificationCache @Inject()(
-  mongoComponent: MongoComponent,
-  appConfig: AppConfig
-)(implicit executionContext: ExecutionContext)
+                                          mongoComponent: MongoComponent,
+                                          appConfig: AppConfig
+                                        )(implicit executionContext: ExecutionContext)
```
Comment (AdamCoops) [inline app/services/NotificationCache.scala:40]: Can we move the implicit or if there is not enough space there closing )

File: app/connectors/DataStoreConnector.scala [hunk 7]
```diff
@@ -29,40 +29,47 @@ import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class DataStoreConnector @Inject()(http: HttpClient,
-                                   metricsReporter: MetricsReporterService)(
-                                    implicit appConfig: AppConfig, ec: ExecutionContext) {
+                                   metricsReporter: MetricsReporterService)
+                                  (implicit appConfig: AppConfig, ec: ExecutionContext) {
```
[the remark this answers is not in the document]
Comment (tyagikumarvineet) [reply app/connectors/DataStoreConnector.scala:33]: yes, it is hence it has been moved to the next line

File: app/controllers/SDESNotificationsController.scala (hunk 1 above)
Comment (tyagikumarvineet) [reply app/controllers/SDESNotificationsController.scala:36]: Done

File: app/controllers/SearchAuthoritiesController.scala (hunk 2 above)
Comment (tyagikumarvineet) [reply app/controllers/SearchAuthoritiesController.scala:30]: Done

File: app/controllers/SubscriptionDisplayRequestController.scala [hunk 8]
```diff
@@ -27,18 +27,21 @@ import javax.inject.Inject
 import scala.concurrent.ExecutionContext
 
 class SubscriptionDisplayRequestController @Inject()(connector: Sub09Connector,
-                                                     cc: ControllerComponents)(implicit ec: ExecutionContext) extends BackendController(cc) {
+                                                     cc: ControllerComponents
+                                                    )(implicit ec: ExecutionContext) extends BackendController(cc) {
```
[the remark this answers is not in the document]
Comment (tyagikumarvineet) [reply app/controllers/SubscriptionDisplayRequestController.scala:31]: Done

File: app/services/NotificationCache.scala (hunk 6 above)
Comment (tyagikumarvineet) [reply app/services/NotificationCache.scala:40]: Done

File: app/controllers/StatementSearchFailureNotificationController.scala (hunk 4 above)
Comment (tyagikumarvineet) [reply app/controllers/StatementSearchFailureNotificationController.scala:146]: Done

File: app/models/responses/StatementSearchFailureNotificationErrorResponse.scala (hunk 5 above)
Comment (tyagikumarvineet) [reply app/models/responses/StatementSearchFailureNotificationErrorResponse.scala:61]: Done, logic has been moved to a separate method

**Review by AdamCoops — APPROVED** [review]

**Pull request merged**